src/v/security/authorizer.h

```diff
@@ -215,15 +216,20 @@ class authorizer final {
     // allow operation when no ACL match is found
     using allow_empty_matches = ss::bool_class<struct allow_empty_matches_type>;
 
-    explicit authorizer(
-      std::function<config::binding<std::vector<ss::sstring>>()> superusers_cb)
-      : authorizer(allow_empty_matches::no, superusers_cb) {}
+    authorizer() = delete;
+
+    authorizer(
+      config::binding<std::vector<ss::sstring>> superusers,
+      const role_store& roles)
```

Comment:
optional nit: I prefer to take the pointer here too so it's more obvious at the callsite that it isn't a copy.

Reply:
>more obvious at the callsite that it isn't a copy

great point

Reply:
done